server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java

```diff
@@ -408,7 +408,9 @@ public short partitionRecordVersion() {
     }
 
     public short fetchRequestVersion() {
-        if (this.isAtLeast(IBP_3_7_IV4)) {
+        if (this.isAtLeast(IBP_4_0_IV0)) {
```

Comment:
I can see why you don't want to fix this here (since there is another pending PR that will fix it). But given that 4.0 is the wrong version, we should have a comment at least.

Reply:
I fiiled this issue to track this [Metadata version 3.9 should return Fetch version 17](https://issues.apache.org/jira/browse/KAFKA-17018)

I'll also add a comment here referencing the Jira.

Reply:
we will fix this in https://github.com/apache/kafka/pull/16347/commits